util/predicates/cluster_predicates.go

```diff
@@ -341,3 +341,45 @@ func processIfTopologyManaged(scheme *runtime.Scheme, logger logr.Logger, object
 	logger.V(6).Info("Cluster does not have topology, blocking further processing")
 	return false
 }
+
+// ClusterTopologyVersionChanged returns a Predicate that returns true when cluster.Spec.Topology.Version
+// was changed.
+func ClusterTopologyVersionChanged(scheme *runtime.Scheme, logger logr.Logger) predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if gvk, err := apiutil.GVKForObject(e.ObjectOld, scheme); err == nil {
```

Comment:
```suggestion
			logger := logger.WithValues("predicate", "ClusterTopologyVersionChanged", "eventType", "update")
			if gvk, err := apiutil.GVKForObject(e.ObjectOld, scheme); err == nil {
```

Reply:
cl.assical issue 🤦 thx!

Reply:
:D My primary mode of review nowadays is "compare with similar stuff"